sql/core/src/main/scala/org/apache/spark/sql/execution/SubqueryBroadcastExec.scala

```diff
@@ -89,10 +90,15 @@ case class SubqueryBroadcastExec(
         val proj = UnsafeProjection.create(expr)
         val keyIter = iter.map(proj).map(_.copy())
 
-        val rows = keyIter.toArray[InternalRow].distinct
+        val rows = if (broadcastRelation.keyIsUnique) {
+          keyIter.toArray[InternalRow]
+        } else {
+          keyIter.toArray[InternalRow].distinct
+        }
         val beforeBuild = System.nanoTime()
         longMetric("collectTime") += (beforeBuild - beforeCollect) / 1000000
         val dataSize = rows.map(_.asInstanceOf[UnsafeRow].getSizeInBytes).sum
+        longMetric("numOutputRows") += rows.length
```

Comment:
Out of curiosity, why does this metric also have to be added / updated?

Reply:
The number of rows may be changed:
<img width="282" alt="image" src="https://user-images.githubusercontent.com/5399861/160506147-a70a88b7-5da9-4088-bc32-f81a21b95fec.png">